insights/combiners/redhat_release.py

```diff
@@ -42,23 +40,94 @@ def deserialize(_type, obj, root=None):
 
 @combiner([rht_release, Uname])
 def redhat_release(rh_release, un):
-    """Check uname and redhat-release for rhel major/minor version.
+    """
+    Combiner method to check uname and redhat-release for rhel major/minor
+    version.
 
     Prefer uname to redhat-release.
 
     Returns:
-        Release: A named tuple with `major` and `minor` version
-        components.
+        Release: A named tuple with the following items:
+            - major: integer
+            - minor: integer
+            - rhel: string
 
     Raises:
-        Exeption: If the version can't be determined even though a Uname
+        ParseException: If the version can't be determined even though a Uname
             or RedhatRelease was provided.
+
+    Examples:
+        >>> rh_release.major
+        7
+        >>> rh_release.minor
+        2
+        >>> rh_release.rhel
+        '7.2'
+        >>> rh_release
+        Release(major=7, minor=2, rhel='7.2')
+
     """
 
     if un and un.release_tuple[0] != -1:
-        return Release(*un.release_tuple)
+        return Release(
+                un.redhat_release.major,
+                un.redhat_release.minor,
+                '{0}.{1}'.format(un.redhat_release.major, un.redhat_release.minor))
 
     if rh_release:
-        return Release(rh_release.major, rh_release.minor)
+        return Release(rh_release.major, rh_release.minor, rh_release.version)
+
+    raise ParseException("Unabled to determine release.")
+
+
+@combiner([Uname, rht_release])
+class RedHatRelease(object):
```

Comment:
Although we don't currently use them, this class also needs a serializer/deserializer.  See this code for an example of these for a class: https://github.com/RedHatInsights/insights-core/blob/master/insights/tests/test_serde.py#L32

Reply:
@bfahr - Done. However I didn't involve the three `rhel[6-8]` attributes in the serializer and deserializer, please check if this is OK. Thanks.